src/api/telemetry/TelemetryAPI.js

```diff
@@ -251,69 +251,50 @@ export default class TelemetryAPI {
   }
 
   /**
-   * Canonicalizes options object by creating a standardized version with sorted keys
-   * and recursively processed values. This ensures consistent object representation
-   * regardless of property order.
+   * Sanitizes objects for consistent serialization by:
+   * 1. Converting non-plain objects (class instances) and functions to string markers (e.g. "@keyName")
+   * 2. Sorting object keys alphabetically to ensure consistent ordering
+   * 3. Recursively processing nested objects and arrays
    *
-   * Used internally to generate consistent hashes for caching subscription options.
+   * This is primarily used to generate consistent hash values for telemetry subscription options,
+   * ensuring that equivalent options objects produce the same hash regardless of property order
+   * or the presence of non-serializable values. While this would normally be a private method,
+   * it is exposed for use in JSON.stringify.
    *
-   * @private
-   * @param {Object|Array|*} options The options object to canonicalize
-   * @returns {Object|Array|*} A canonicalized version of the input where object keys are sorted
-   */
-  #canonicalizeOptions(options) {
-    if (typeof options !== 'object' || options === null) {
-      return options;
-    }
-
-    if (Array.isArray(options)) {
-      return options.map((item) => this.#canonicalizeOptions(item));
-    }
-
-    return Object.keys(options)
-      .sort()
-      .reduce((acc, key) => {
-        acc[key] = this.#canonicalizeOptions(options[key]);
-        return acc;
-      }, {});
-  }
-
-  /**
-   * Filters out class instances and functions from an object, returning a plain
-   * data object suitable for JSON serialization.
-   *
-   * @private
+   * @param {string} key The current property key being processed
    * @param {Object|Array|*} value The value to sanitize
-   * @returns {Object|Array|*} A sanitized copy with class instances and functions removed
+   * @returns {Object|Array|string|*} The sanitized value:
+   *    - Plain objects: A new object with sorted keys and sanitized values
+   *    - Arrays: A new array with sanitized elements
+   *    - Functions/Class instances: String in format "@keyName"
+   *    - Primitives: Returned as-is
    */
-  #sanitizeForSerialization(value) {
-    if (value === null || value === undefined) {
+  sanitizeForSerialization(key, value) {
+    // Handle null and primitives directly
+    if (value === null || typeof value !== 'object') {
       return value;
     }
 
-    // Handle primitive types, arrays and objects
-    if (typeof value !== 'object') {
-      return typeof value !== 'function' ? value : undefined;
-    }
-
+    // Handle arrays
     if (Array.isArray(value)) {
-      return value.map((item) => this.#sanitizeForSerialization(item));
+      return value.map((item, index) => this.sanitizeForSerialization(String(index), item));
```

Comment:
Just return the array itself, the replacer function (sanitizeForSerialization) will automatically be called for each member. see docs - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#the_replacer_parameter